### ISSUE TITLE
provide parameters for account name, key and container name

### ISSUE THREAD

**Issue by andyzhangx** [Azure/azure-storage-fuse#32]
As I could see, "account name, key and container name" are all in connection.cfg, is it possible for keep it in memory, e.g. like what mount cifs does for azure file:
```
sudo mount -t cifs //pvc989495346002.file.core.windows.net/sharename [mount_path] -o vers=3.0,username=pvc989495346002,password=...,dir_mode=0777,file_mode=0777,sec=ntlmssp
```
Store account key in file is not acceptable in some production scenario.

**Comment by andyzhangx** [comment]
@seguler

**Comment by seguler** [comment]
Why can't you restrict access to connection.cfg instead ? Such as 400 permissions for the owner to read. If we put this as an option on the command line; the arguments including the key will be visible in output of ps - ef which will not be any more secure than having a connection file which is only accessible by the user.

**Comment by AlexanderYukhanov** [comment]
As Sercan mentioned command line arguments are not protected at all. the following command will report r--r--r-- for all command lines:
sudo find /proc  -name cmdline -exec ls -l {} \;

**Comment by andyzhangx** [comment]
@AlexanderYukhanov 
If you take a look at `mount -t cifs` command for azure file service, you will find it's using parameter way
```
mount -t cifs //pvc989495346002.file.core.windows.net/andy-k8s182-dynamic-pvc-d69ed1b1-dc06-11e7-8ffc-000d3a019e90 /mntx -o vers=3.0,username=pvc989495346002,password=...,dir_mode=0777,file_mode=0777,sec=ntlmssp
```

And then if you `sudo vi /proc/mounts`, you will find `password` is not displayed in proc
```
//pvc989495346002.file.core.windows.net/andy-k8s182-dynamic-pvc-d69ed1b1-dc06-11e7-8ffc-000d3a019e90 /mntx cifs rw,relatime,vers=3.0,sec=ntlmssp,cache=strict,username=pvc989495346002,domain=,uid=0,noforceuid,gid=0,noforcegid,addr=52.239.157.200,file_mode=0777,dir_mode=0777,persistenthandles,nounix,serverino,mapposix,rsize=1048576,wsize=1048576,echo_interval=60,actimeo=1 0 0
```

In kubernetes, the secret is stored in master ectd db and only permitted user (RBAC) could access secret and it will be used only once, the secret will not be stored in any local file. If the secret is in local file, it will break RBAC scenario. 
I am not sure whether fuse driver could do that(not display password?) since it's not using standard `mount` command.

**Comment by AlexanderYukhanov** [comment]
Not related to the issue itself but providing a password to the mount command is still unsafe if there are chances that there is a user session on the vm. The user may do something like:
$ while true; do ps -afx | grep "mount -t cifs" | grep -v grep ; done
and capture the mount command line.

**Comment by AlexanderYukhanov** [comment]
If such security feature required I would suggest blobfuse to be able to read the password from the input stream.

**Comment by AlexanderYukhanov** [comment]
And just for the info, blobfuse process has the key value in its address space, so root can always attach a debugger and get it.

**Comment by andyzhangx** [comment]
@AlexanderYukhanov in k8s, the "mount -t cifs" happens on a container called kubelet, there is some security isolation since it's a container. While if user is root, user can get anything anyway.

**Comment by seguler** [comment]
Our agreement after a few e-mail exchanges is to provide a way to fetch these values from environment variables.

**Comment by andyzhangx** [comment]
Thanks guys!

**Comment by seguler** [comment]
@andyzhangx we now have enabled environment variable support in the dev branch. Let us know what you think about the implementation. Account name and key is retrieved from env variables, whereas --container-name is retrieved from the command line arguments.

**Comment by seguler** [comment]
This is now released in blobfuse 0.2.4
